nvflare/app_common/logging/log_receiver.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_constant import ReturnCode, StreamCtxKey, SystemComponents
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.logging.constants import Channels
+from nvflare.app_common.streamers.file_streamer import FileStreamer
+from nvflare.widgets.widget import Widget
+
+
+class LogReceiver(Widget):
+    def __init__(self):
+        """Receives log data.
+
+        If adding additional log types, make sure nvflare.apis.storage.ComponentPrefixes has
+        the corresponding log type.
+        """
+        super().__init__()
+
+    def process_log(self, stream_ctx: dict, fl_ctx: FLContext):
```

Comment:
Change type hint here: stream_ctx: dict to. stream_ctx: StreamContext

You need to:
from nvflare.api.streaming import StreamContext